containers/unit_tests/TestBitset.hpp

```diff
@@ -237,6 +242,44 @@ void test_bitset() {
 }
 
 TEST(TEST_CATEGORY, bitset) { test_bitset<TEST_EXECSPACE>(); }
+
+TEST(TEST_CATEGORY, bitset_default_constructor_no_alloc) {
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableAllocs());
+
+  auto success = validate_absence(
+      [&]() {
+        Kokkos::Bitset bs;
+        EXPECT_FALSE(bs.is_allocated());
+      },
+      [&](AllocateDataEvent) {
+        return MatchDiagnostic{true, {"Found alloc event"}};
+      });
+  ASSERT_TRUE(success);
+
+  listen_tool_events(Config::DisableAll());
+}
+
+TEST(TEST_CATEGORY, bitset_zero_size_does_alloc) {
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableAllocs());
+
+  auto success = validate_existence(
+      [&]() {
+        Kokkos::Bitset bs1(0);
+        EXPECT_TRUE(bs1.is_allocated());
+
+        Kokkos::Bitset bs2(Kokkos::view_alloc("MyBitset"), 0);
+        EXPECT_TRUE(bs2.is_allocated());
+      },
+      [&](AllocateDataEvent) {
+        return MatchDiagnostic{true, {"Found alloc event"}};
+      });
+  ASSERT_TRUE(success);
```

Comment:
Shouldn't this test be done in 2 distinct `validate_existence` ? If I understand it correctly, in its current state, the first allocation `bs1(0)` could in fact not allocate anything and the test would still pass if `bs2` allocates something?

Not requesting any change though, I just wanted to make sure I understand `validate_existence` :smile:

BTW is there any helper in the `Kokkos` testing toolbox that would not only allow us to ensure that some event occurred (in this case `AllocateDataEvent`) but also to count them ?

Reply:
You are correct and that is how I had written the test initially but I didn't think it was worth the boiler plate code.

I don't know that there is currently something to count the number of allocations even though writing directly a tool that does that would be trivial.